src/main/java/io/gravitee/common/http/HttpHeaders.java

```diff
@@ -475,6 +476,15 @@ public void setAll(Map<String, String> values) {
         return singleValueMap;
     }
 
+    @Override
+    public boolean containsAllKeys(Collection<String> keys) {
+        final LinkedCaseInsensitiveMap<Object> headers = (LinkedCaseInsensitiveMap) this.headers;
```

Comment:
I think it is better to override the equals method inside this class as well to avoid any weird issues LinkedCaseInsensitiveMap